src/d/actor/d_a_obj_doguu.cpp

```diff
@@ -5,120 +5,811 @@
+        }   
+    }
+    return phase_state;
 }
 
+bool daObjDoguu_c::_delete(){
+    dKy_plight_cut(&mLightInfluence);
+    dComIfG_resDelete(&mPhs,"Doguu");
+    return true;
+}
+
+
 /* 00001CC8-00001D0C       .text daObjDoguu_Delete__FPv */
-static BOOL daObjDoguu_Delete(void*) {
-    /* Nonmatching */
+static BOOL daObjDoguu_Delete(void* i_this) {
+    return static_cast<daObjDoguu_c*>(i_this)->_delete();
 }
 
 /* 00001D0C-00001D30       .text daObjDoguu_Draw__FPv */
-static BOOL daObjDoguu_Draw(void*) {
-    /* Nonmatching */
+static BOOL daObjDoguu_Draw(void* i_this) {
+    return static_cast<daObjDoguu_c*>(i_this)->_draw();
 }
 
 /* 00001D30-00001F64       .text _draw__12daObjDoguu_cFv */
 bool daObjDoguu_c::_draw() {
-    /* Nonmatching */
+    g_env_light.settingTevStruct(TEV_TYPE_ACTOR, &current.pos, &tevStr);
+    g_env_light.setLightTevColorType(field_0x6CC, &tevStr);
+    g_env_light.setLightTevColorType(field_0x6D0, &tevStr);
+    g_env_light.setLightTevColorType(field_0x6D4, &tevStr);
+    g_env_light.setLightTevColorType(field_0x6D8, &tevStr);
+    
+    setEffectMtx(&eyePos, attr().mScale);
+
+    if(field_0x8A1){
+        J3DMaterialTable* bmt = (J3DMaterialTable*)dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bmt_vgsb[field_0x894]);
+        
+        field_0x6D4->getModelData()->setMaterialTable(bmt, J3DMatCopyFlag_All);
+        mBck1.entry(field_0x6D0->getModelData());
+        mDoExt_modelUpdateDL(field_0x6D0);
+        field_0x6D0->getModelData()->getJointNodePointer(0)->setMtxCalc(NULL);
+
+        mBck2.entry(field_0x6D4->getModelData());
+        mDoExt_modelUpdateDL(field_0x6D4);
+        field_0x6D4->getModelData()->getJointNodePointer(0)->setMtxCalc(NULL);
+    } else {
+        J3DMaterialTable* bmt = (J3DMaterialTable*)dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bmt_vgsm[field_0x894]);
+        field_0x6CC->getModelData()->setMaterialTable(bmt, J3DMatCopyFlag_All);
+        mBrk.entry(field_0x6CC->getModelData());
+        mDoExt_modelUpdateDL(field_0x6CC);
+        field_0x6CC->getModelData()->getMaterialTable().removeTevRegAnimator(mBrk.getBrkAnm());
+    }
+    if (field_0x8A0 == true) {
+        mBck3.entry(field_0x6D8->getModelData());
+        mDoExt_modelUpdateDL(field_0x6D8);
+        field_0x6D8->getModelData()->getJointNodePointer(0)->setMtxCalc(NULL);       
+    }
+    return true;
 }
 
 /* 00001F64-00002264       .text setEffectMtx__12daObjDoguu_cFPC4cXyzf */
-void daObjDoguu_c::setEffectMtx(const cXyz*, float) {
-    /* Nonmatching */
+void daObjDoguu_c::setEffectMtx(const cXyz* i_pos, float i_scale) {
+    static Mtx mtx_adj = { 
+        0.5f, 0.0f, 0.0f, 0.5f,
+        0.0f, -0.5f, 0.0f, 0.5f,
+        0.0f, 0.0f, 1.0f, 0.0f,
+    };
+    
+    float scale = 1.0f /i_scale;
+    camera_class* camera = dCam_getCamera();
+    cXyz lookDir = *i_pos - camera->mLookat.mEye;
+    cXyz lightDir;
+    cXyz refl;
+
+    dKyr_get_vectle_calc((cXyz*)&tevStr.mLightPosWorld, (cXyz*)i_pos, &lightDir);
```

Comment:
I think the first cast is unnecessary?
```suggestion
    dKyr_get_vectle_calc(&tevStr.mLightPosWorld, (cXyz*)i_pos, &lightDir);
```